src/olympia/reviews/views.py

```diff
@@ -327,22 +327,46 @@ class ReviewViewSet(AddonChildMixin, ModelViewSet):
 
     queryset = Review.objects.all()
 
+    def set_addon_object_from_review(self, review):
+        """Set addon object on the instance from a review object."""
+        self.kwargs['addon_pk'] = str(review.addon.pk)
+        del self.addon_object  # Force get_addon_object() to try again.
```

Comment:
`None` is a specific state, yeah - in fact, in both methods that call this, `reply` and `flag`, `addon_object` is initially set to `None` (because it's not passed in query string/post data for those views) and that's why we have to reset it.